packages/react-core/src/demos/DataList/examples/DataListPagination.tsx

```diff
@@ -0,0 +1,187 @@
+    endIdx: number | undefined
+  ) => {
+    setPaginatedRows(rows.slice(startIdx, endIdx));
+    setPage(newPage);
+    setPerPage(newPerPage);
+  };
+
+  const renderPagination = (id: string, variant: PaginationVariant, isCompact: boolean, isSticky: boolean) => (
+    <Pagination
+      id={id}
+      variant={variant}
+      itemCount={rows.length}
+      page={page}
+      perPage={perPage}
+      isCompact={isCompact}
+      isSticky={isSticky}
+      onSetPage={handleSetPage}
+      onPerPageSelect={handlePerPageSelect}
+      titles={{
+        paginationAriaLabel: `${variant} pagination`
+      }}
+    />
+  );
+
+  const toolbarItems = (
+    <React.Fragment>
+      <ToolbarItem variant="bulk-select">
+        <MenuToggle
+          aria-label="Select data list items"
+          splitButtonOptions={{
+            items: [
+              <MenuToggleCheckbox
+                id="split-dropdown-checkbox"
+                key="split-dropdown-checkbox"
+                aria-label={'Select all data list items'}
+              />
+            ]
+          }}
+        ></MenuToggle>
+      </ToolbarItem>
+      <ToolbarItem>
+        <MenuToggle>Filter by creator name</MenuToggle>
+      </ToolbarItem>
+      <ToolbarItem variant="overflow-menu">
+        <OverflowMenu breakpoint="md">
+          <OverflowMenuItem>
+            <Button variant="primary">Create instance</Button>
+          </OverflowMenuItem>
+          <OverflowMenuControl hasAdditionalOptions>
+            <MenuToggle aria-label="Toolbar kebab overflow menu" variant="plain">
+              <EllipsisVIcon />
+            </MenuToggle>
+          </OverflowMenuControl>
+        </OverflowMenu>
+      </ToolbarItem>
+      <ToolbarItem variant="pagination" align={{ default: 'alignRight' }}>
+        {renderPagination('top-pagination', PaginationVariant.top, false, false)}
+      </ToolbarItem>
+    </React.Fragment>
+  );
+
+  return (
+    <DashboardWrapper mainContainerId="main-content-datalist-view-pagination" breadcrumb={null}>
+      <PageSection variant={PageSectionVariants.light}>
+        <TextContent>
+          <Text component="h1">Projects</Text>
+          <Text component="p">This is a demo that showcases PatternFly Data List</Text>
+        </TextContent>
+      </PageSection>
+      <PageSection isFilled>
+        <Toolbar id="toolbar-group-types">
+          <ToolbarContent>{toolbarItems}</ToolbarContent>
+        </Toolbar>
+        <DataList aria-label="Demo data list">
+          {paginatedRows.map((row, rowIndex) => {
+            const { name, threads, applications, workspaces, lastModified } = row;
+            return (
+              <DataListItem aria-labelledby={`Demo-item-${rowIndex}`} key={rowIndex}>
+                <DataListItemRow>
+                  <DataListItemCells
+                    dataListCells={[
+                      <DataListCell isFilled={false} key="buttons1">
+                        <Flex direction={{ default: 'column' }}>
```

Comment:
This is just a nit, I'm not sure how close we need to match the core demo, especially this part which is just content and doesn't impact the purpose of the demo (static bottom pagination) but if we do want to fix it, the layout is below. One decent visual thing it will address is to add space between the icon/text pairings in each row. You could also just add a `{' '}` or something there, too, or get the svg and number to render on the same line like the [basic data list demo](https://github.com/patternfly/patternfly-react/blob/1f492d118b31f73e347d17f5ee8911b34d79c89f/packages/react-core/src/demos/DataList/examples/DataListBasic.tsx#L107-L115)

```
flex.column.space-items-md
  flex.column.space-items-none
    flex-item
      // <p> text
    flex-item
      // <small> text
  flex.wrap
    flex.space-items-xs // one of these for each svg/number pair
      // <svg> and number
    flex-item
      // "Updated [n] days ago" text
```